### ISSUE TITLE
1 package has known vulnerabilities, guzzlehttp/psr7 (2.1.0)

### ISSUE THREAD

**Issue by JackPotte** [knpuniversity/oauth2-client-bundle#366]
symfony security:check --disable-exit-code
Symfony Security Check Report
=============================

1 package has known vulnerabilities.

guzzlehttp/psr7 (2.1.0)
-----------------------

 * [CVE-2022-24775][]: Inproper parsing of HTTP headers

[CVE-2022-24775]: https://github.com/guzzle/psr7/security/advisories/GHSA-q7rv-6hp3-vh

```

./composer.phar depends guzzlehttp/psr7
guzzlehttp/guzzle  7.4.1  requires  guzzlehttp/psr7 (^1.8.3 || ^2.1)  

./composer.phar depends guzzlehttp/guzzle
league/oauth2-client  2.6.1  requires  guzzlehttp/guzzle (^6.0 || ^7.0)  

./composer.phar depends league/oauth2-client
knpuniversity/oauth2-client-bundle  v2.10.0  requires  league/oauth2-client (^2.0)
```

**Comment by bocharsky-bw** [comment]
Thanks for sharing this.

Seems it was fixed upstream in guzzlehttp/psr7 already, right? I see the latest version of guzzlehttp/psr7 v2.2.1. So you just need to upgrade your project dependencies to the latest, in particular, make sure guzzlehttp/psr7 package was upgraded to a newer version, i.e. >2.1.0

**Comment by JackPotte** [comment]
Actually I've launched:
`composer update knpuniversity/oauth2-client-bundle -W`
but it wasn't enough. Then :
```
composer require guzzlehttp/psr7
composer remove guzzlehttp/psr7
```
and finally the version depending on knpuniversity/oauth2-client-bundle was updated.

**Comment by bocharsky-bw** [comment]
Hm, I think just `composer update guzzlehttp/psr7`  would help too, you even didn't have to update `knpuniversity/oauth2-client-bundle` because the problem is upstream

**Comment by jrushlow** [comment]
`composer update` should bring you up to date w/ the patched psr7 client. The cve was fixed in `guzzlehttp/psr7` `1.8.4` && `2.1.1`

Per the dependency tree - 
```php
// knpuniversity/oauth2-client-bundle
$require = [
    'league/oauth2-client' =>  [
        '^2.0' => [
            'guzzlehttp/guzzle' => [
                '^6.0 || ^7.0' => [
                    'guzzlehttp/psr7' => [
                        '^1.8.3 || ^2.1',
                    ],
                ],
            ],
        ],
    ],
];
```

Even w/ guzzle 6, the `guzzlehttp/psr7` requirement is `^1.6.1`. Essentially that means there are not any dependencies blocking the patched client.

Let us know @JackPotte if you have any more issues getting things updated.
